examples/systemcalls/systemcalls.c

```diff
@@ -34,66 +43,92 @@ bool do_system(const char *cmd)
 *   by the command issued in @param arguments with the specified arguments.
 */
 
-bool do_exec(int count, ...)
+bool do_exec(int count, ...) 
 {
+    if (count < 1) 
+    {
+        return false;
+    }
+
     va_list args;
     va_start(args, count);
-    char * command[count+1];
-    int i;
-    for(i=0; i<count; i++)
+    char *command[count + 1];
+
+    for (int i = 0; i < count; i++) 
     {
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
-
-/*
- * TODO:
- *   Execute a system command by calling fork, execv(),
- *   and wait instead of system (see LSP page 161).
- *   Use the command[0] as the full path to the command to execute
- *   (first argument to execv), and use the remaining arguments
- *   as second argument to the execv() command.
- *
-*/
-
     va_end(args);
 
-    return true;
+    pid_t pid = fork();
+    if (pid == -1) {
+        return false;
+    } else if (pid == 0) 
+    {
+        // Child process: Execute the command
+        execv(command[0], command);
+        _exit(EXIT_FAILURE);  // execv failed
+    }
+
+    // Parent process: Wait for the child to finish
+    int status;
+    if (waitpid(pid, &status, 0) == -1) 
+    {
+        return false;
+    }
+
+    return WIFEXITED(status) && WEXITSTATUS(status) == 0;
 }
 
 /**
 * @param outputfile - The full path to the file to write with command output.
 *   This file will be closed at completion of the function call.
 * All other parameters, see do_exec above
 */
-bool do_exec_redirect(const char *outputfile, int count, ...)
+bool do_exec_redirect(const char *outputfile, int count, ...) 
 {
+    if (count < 1 || outputfile == NULL) 
+    
+    {
+        return false;
+    }
+
     va_list args;
     va_start(args, count);
-    char * command[count+1];
-    int i;
-    for(i=0; i<count; i++)
-    {
+    char *command[count + 1];
+
+    for (int i = 0; i < count; i++) {
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
+    va_end(args);
 
+    pid_t pid = fork();
```

Comment:
@Sriramz2002 think about how you could avoid duplicating logic with the function above.